vmdk_plugin/drivers/photon/photon_driver.go

```diff
@@ -373,22 +389,26 @@ func (d *VolumeDriver) processMount(r volume.MountRequest) volume.Response {
 		return volume.Response{Mountpoint: d.getMountPoint(r.Name)}
 	}
 
-	// There can be redundant mounts till refcounts are properly initialized
-	// TODO: #1220
-	status, err := d.GetVolume(r.Name)
-	if err != nil {
-		d.decrRefCount(r.Name)
-		return volume.Response{Err: err.Error()}
+	if plugin_utils.AlreadyMounted(r.Name, d.mountRoot) {
+		log.WithFields(log.Fields{"name": r.Name}).Info("Already mounted, skipping mount. ")
+		return volume.Response{Mountpoint: d.getMountPoint(r.Name)}
+	}
+
+	// get volume metadata if required
+	if volumeMeta == nil {
+		if volumeMeta, err = d.GetVolume(r.Name); err != nil {
+			return volume.Response{Err: err.Error()}
+		}
 	}
 
-	fstype, exists := status[fsTypeTag]
+	fstype, exists := volumeMeta[fsTypeTag]
 	if !exists {
 		fstype = fs.FstypeDefault
 	}
 
 	skipAttach := false
 	// If the volume is already attached to the VM, skip the attach.
-	if state, stateExists := status["State"]; stateExists {
+	if state, stateExists := volumeMeta["State"]; stateExists {
 		if strings.Compare(state.(string), "DETACHED") != 0 {
```

Comment:
This is not your change but please file an issue and let's revisit.
Generally we don't make decision based on data in KV file and definitely not inside guest. Disk may be attached to some other VM. If neither refCnt nor mount map suggest disk is attached to this VM then we must issue attach to ESX. If it is already attached, it is a no-op from ESX perspective anyway.